src/toniarts/openkeeper/game/state/loading/LoadingState.java

```diff
@@ -74,6 +79,9 @@ public void initialize(final AppStateManager stateManager, final Application app
 
         // Load up the title screen
         setupTitleScreen();
+
+        loadingThread = new LoadingThread(this.name);
+        loadingThread.start();

```

Comment:
The problem with this.... While it is quite ugly to just create a new thread when you create this, but the loading will start immediately. Now instead the loading is governed by the framerate. Well, no really that big of an issue. But what if you load the app minimized or alt tab away. Loading basically never starts because it waits for the screen to draw.

Reply:
This is not a bug, this is feature ;-). I can combine our code. Pass name and app to the constructor. Combine?

Reply:
Yeah the other changes I think are just fine. I would be conservative about the thread starting yes. So yes :)